app/admin/service.rb

```diff
@@ -73,6 +73,13 @@
     end
   end
 
+  form do |f|
+    f.object.natural_key ||= SecureRandom.hex(2)
+    f.object.publish_token ||= SecureRandom.hex(64)
```

Comment:
I don't think the publish token should be user editable. It should already be [set when it's created](https://github.com/alphagov/gsd-api/blob/dc1173b1b574d3ed0ca1e1ab1318289022023313/app/models/service.rb#L11-L13).